server/server.go

```diff
@@ -76,13 +76,14 @@ func NewServer(cfg Config, e *sqle.Engine, sb SessionBuilder, listener ServerEve
 	}
 
 	listenerCfg := mysql.ListenerConfig{
-		Listener:           l,
-		AuthServer:         e.Analyzer.Catalog.MySQLDb,
-		Handler:            handler,
-		ConnReadTimeout:    cfg.ConnReadTimeout,
-		ConnWriteTimeout:   cfg.ConnWriteTimeout,
-		MaxConns:           cfg.MaxConnections,
-		ConnReadBufferSize: mysql.DefaultConnBufferSize,
+		Listener:                 l,
+		AuthServer:               e.Analyzer.Catalog.MySQLDb,
+		Handler:                  handler,
+		ConnReadTimeout:          cfg.ConnReadTimeout,
+		ConnWriteTimeout:         cfg.ConnWriteTimeout,
+		MaxConns:                 cfg.MaxConnections,
+		ConnReadBufferSize:       mysql.DefaultConnBufferSize,
+		AllowClearTextWithoutTLS: true,
```

Comment:
I don't really like this setting in general...but if we need it, it definitely shouldn't be default / it should only be used for tests? Is there a way we can get away without it? I don't see it referenced in the Vitess or the GMS code, but I might be missing something...